docs/design/y-flow/y-flow-nb-api.md

```diff
@@ -0,0 +1,350 @@
+passing traffic.
+
+## Update operation
+
+Changes will be propagated on all already existing subordinate flows.
+
+REST endpoints: 
+ * `PUT /v2/y-flows/{y_flow_id}`
+ * `PATCH /v2/y-flows/{y_flow_id}`
+
+Request/response bodies are the same to the create operation. In case of PATCH request only present in request body
+fields will be updated.
+
+## Reroute operation
+
+Changes will be propagated on all already existing subordinate flows.
+
+REST endpoint: `POST /v2/y-flows/{y_flow_id}/reroute`
+
+Request body is ignored, response payload:
+
+```json
+{
+  "rerouted": true,
+  "shared_path": {
+    "nodes": [
+      {
+        "port_no": 0,
+        "segment_latency": 10,
+        "switch_id": "00:00:00:00:00:00:00:04"
+      }
+    ]
+  },
+  "sub_flow_paths": [
+    {
+      "flow_id": "fAAAAAAAAAAAAAAA",
+      "nodes": [
+        {
+          "port_no": 0,
+          "segment_latency": 0,
+          "switch_id": "00:00:00:00:00:00:00:05"
+        }
+      ]
+    },
+    {
+      "flow_id": "fAAAAAAAAAAAAAAB",
+      "nodes": [
+        {
+          "port_no": 0,
+          "segment_latency": 0,
+          "switch_id": "00:00:00:00:00:00:00:06"
+        }
+      ]
+    }
+  ]
+}
+```
+
+## Delete operation
+
+REST endpoint: `DELETE /v2/y-flows/{y_flow_id}`
+
+Request body is ignored, response body is the same to the read operation.
+
+## Read subordinate flows
+
+REST endpoint: `GET /v2/y-flows/{y_flow_id}/flows`
+
+Response body:
+
+```json
+{
+  "flows": [
+    {
+      "flow_id": "fAAAAAAAAAAAAAAB",
+      "endpoint": {
+        "switch_id": "00:00:00:00:00:00:00:03",
+        "port_number": 0,
+        "vlan_id": 0,
+        "inner_vlan_id": 0
+      },
+      "shared_endpoint": {
+        "vlan_id": 0,
+        "inner_vlan_id": 0
+      },
+      "status": "status",
+      "description": "description",
+      "time_create": "YYYY-MM-DDThh:mm:ss[.SSS]",
+      "time_update": "YYYY-MM-DDThh:mm:ss[.SSS]"
+    }, {
+      "flow_id": "fAAAAAAAAAAAAAAB",
+      "endpoint": ...
+    }
+  ]
+}
+```
+
+## Read paths
+
+REST endpoint: `GET /v2/y-flows/{y_flow_id}/paths`
+
+Response body:
+
+```json
+{
+  "shared_path": {
+    "nodes": [
+      {
+        "port_no": 0,
+        "segment_latency": 0,
+        "switch_id": "00:00:00:00:00:00:00:08"
+      }
+    ]
+  },
+  "sub_flow_paths": [
+    {
+      "flow_id": "fAAAAAAAAAAAAAAA",
+      "nodes": [
+        {
+          "port_no": 0,
+          "segment_latency": 0,
+          "switch_id": "00:00:00:00:00:00:00:09"
+        }
+      ]
+    },
+    {
+      "flow_id": "fAAAAAAAAAAAAAAB",
+      "nodes": [
+        {
+          "port_no": 0,
+          "segment_latency": 0,
+          "switch_id": "00:00:00:00:00:00:00:09"
+        }
+      ]
+    }
+  ]
+}
+```
+
+# Existing API changes
+
+All flows CRUD operation responses will be extended with following fields:
+
+```json
+{
+  "...": "...",
+  "y_flow_id": "yf-AAAAAAAAAAAAAAA",
+  "....": "..."
+}
```

Comment:
How about `v1/flows/:flow-id/validate` or `v1/flows/:flow-id/ping` endpoints? should we extend them for y-fow?

Reply:
Yes, we will support validation and ping for sub-flows.